.github/workflows/ci.yml

```diff
@@ -315,3 +382,31 @@ jobs:
           name: "trivy-results-nginx-kubernetes-gateway.sarif"
           path: "trivy-results-nginx-kubernetes-gateway.sarif"
         if: always()
+
+  publish-helm:
+    name: Package and Publish Helm Chart
+    runs-on: ubuntu-22.04
+    needs: [vars, helm-tests]
+    if: ${{ github.event_name == 'push' && ! startsWith(github.ref, 'refs/heads/release-') }}
```

Comment:
Does this mean we publish on push events to non-release branches?

Reply:
Yes, which ends up only being the main branch - we don't run CI on non-main/ release branches unless the event is a PR. The majority of the "push" events is a merge of a PR to main, and occasionally release branches, which this is skipping.